### PR TITLE
Add delocate

### PR DESCRIPTION
**Pull request by jschueller**
Fixes https://github.com/conda-forge/staged-recipes/issues/5316

### CODE REVIEW

**Comment by msarahan** [comment]
This seems fine, but I'm curious what the purpose of having it on conda-forge is.  It's part of the whole wheel world, which typically runs headlong against the conda way to do things.  What were your thoughts on adding this here?

**Comment by jakirkham** [comment]
Can't speak to @jschueller's reasoning. That said, I can see benefit to adding this.

We already have `auditwheel`, which is used for wheels on Linux. `delocate` is used for wheels on macOS. So not too different.

Besides `conda-build` can already build wheels. If we need to improve that functionality, we may want tools like `auditwheel` and `delocate` around to aid in that build process, which we will need as Conda packages.

File: recipes/delocate/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+test:
+  imports:
+    - delocate
+  commands:
+    - delocate-listdeps -h
+    - delocate-wheel -h
+
+about:
+  home: https://github.com/matthew-brett/delocate
+  license: BSD
```
Comment (jakirkham) [inline recipes/delocate/meta.yaml]: Is this BSD 2-Clause or BSD 3-Clause?

File: recipes/delocate/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+build:
+  number: 0
+  skip: true  # [not osx]
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - pip
+    - python
+    - wheel
```
Comment (jakirkham) [inline recipes/delocate/meta.yaml]: Is `wheel` needed for the build?

File: recipes/delocate/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+source:
+  fn: delocate-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/d/delocate/delocate-{{ version }}.tar.gz
+  sha256: 404819b9f4536903c432e9d99c3d5993ba267f86d73cab51d76cc0d6e04dfd55
+
+build:
+  number: 0
+  skip: true  # [not osx]
+  script: "{{ PYTHON }} -m pip install . -vv"
```
Comment (jakirkham) [inline recipes/delocate/meta.yaml]: May want to add `--no-deps` here. Believe this was fixed in recent versions of `conda-build`, but recall we had pinned an older version of `conda-build` in a few places including `staged-recipes`. So they may not have the fix.

Comment (jschueller) [reply recipes/delocate/meta.yaml]: ok, but i just followed https://conda-forge.org/docs/meta.html#use-pip, so ...

File: recipes/delocate/meta.yaml (hunk 2 above)
Comment (jschueller) [reply recipes/delocate/meta.yaml]: I guess not

File: recipes/delocate/meta.yaml (hunk 1 above)
Comment (jschueller) [reply recipes/delocate/meta.yaml]: ok, fixed

**Comment by jschueller** [comment]
I want to have delocate here to build wheel packages by repackaging conda-forge binaries for openturns: https://github.com/openturns/build-wheels.

File: recipes/delocate/meta.yaml (hunk 3 above)
Comment (jakirkham) [reply recipes/delocate/meta.yaml]: You're absolutely right. We have had some minor setbacks on this front. We would like to get the syntax to this level of simplicity though. Sorry for the trouble.

**Comment by jschueller** [comment]
so, good to merge ?

**Comment by jakirkham** [comment]
For me, yes. I'll wait though to give other people time to respond.

**Comment by jschueller** [comment]
@conda-forge/help-python please review

**Comment by jschueller** [comment]
ping @conda-forge/staged-recipes

**Comment by jschueller** [comment]
thanks @SylvainCorlay

**Pull request merged**